applications/upgradeability-by-proxy.md

```diff
@@ -0,0 +1,202 @@
+  from [Innogy](https://www.eon.de/de/innogy.html), an early investor in
+  Fractal, for more information.
+* In early 2018 we built a prototype of on-chain KYC using the [ERC
+  725](https://github.com/ethereum/eips/issues/725) and [ERC
+  735](https://github.com/ethereum/eips/issues/735) standards. See [our medium
+  post](https://hackernoon.com/first-impressions-with-erc-725-and-erc-735-identity-and-claims-4a87ff2509c9)
+  on this for more information.
+* In late 2018, we built our first on-chain KYC integration with DIDs and
+  verifiable credentials, when working with https://ixo.world/. Once we
+  confirmed that a user controlled a certain address (by asking them to sign a
+  nonce), we would create their DID on their behalf. Once their KYC
+  verification completed, we would ledger the corresponding credential against
+  this DID. See [our Medium post on
+  this](https://medium.com/hackernoon/decentralized-identity-w3c-standards-and-the-ixo-keysafe-69d7e8d6fd7c)
+  for more information.
+* We're currently working with [KILT](https://www.kilt.io/) to deliver a
+  decentralised KYC solution for [Polkadex](https://www.polkadex.trade/). See
+  [KILT's medium post on
+  this](https://kilt-protocol.medium.com/fractal-kilt-protocol-and-polkadex-collaborate-to-provide-decentralised-kyc-solution-for-exchange-d9d0f8c18163)
+  for more information.
+
+
+### Team Code Repos
+
+* https://github.com/trustfractal/ink-upgrade-template
+* https://github.com/trustfractal/ink
+* eventually https://github.com/paritytech/ink, we hope :)
+
+
+### Team LinkedIn Profiles
+
+* https://www.linkedin.com/in/hugopeixoto
+* https://www.linkedin.com/in/jvsantos
+
+
+## Development Roadmap :nut_and_bolt:
+
+### Overview
+
+* **Total Estimated Duration:** 2~3 months
+* **Full-time equivalent (FTE):** 0.5
+* **Total Costs:** DAI 19400
+
+### Milestone 1 Example — Implement Substrate Modules
+
+* **Estimated Duration:** 1 month
+* **FTE:**  0.5
+* **Costs:** DAI 9700
+
+| Number        | Deliverable   | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | Documentation on how to use the smart contract template. This should answer the question "how do I build my contract in a way that lets me upgrade it afterwards?" |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Article/Tutorial | We will write an article or tutorial that explains the work done as part of the grant. |
+| 1. | An Ink! based upgradeable Smart Contract structure | A collection of Ink! smart contracts that be used as a starting point for people that want to have upgradable smart contracts. This collection will contain upgrade logic (including authorization and data migration calls) and some sample methods that users can adapt when developing their own contracts. |
```

Comment:
Are you planning to target the examples from the [ink! repo](https://github.com/paritytech/ink/tree/master/examples/)? Or some other sample contracts?

Reply:
The goal is to pick a contract that is not too complicated, so that the upgradeability components don't get lost in the code, but not trivial enough that there's nothing to upgrade.

My idea was to get some inspiration from the ink! repository examples and the examples in https://substrate.dev/substrate-contracts-workshop and design something based on that.